### ISSUE TITLE
2021 prelude migration lint: Check if trait is actually implemented

### ISSUE THREAD

**Issue by m-ou-se** [rust-lang/rust#86902]
(As I mentioned [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/268952-edition-2021/topic/'Manual.20crater.20run'.20results/near/244166619):)

I tried out the migration lints on some of my own code, and ran into one false positive:

- A `StructOpt` struct being used with `Opt::from_iter`
```
 help: disambiguate the associated function: `<Opt as StructOpt>::from_iter`
```

This doesn't actually become ambiguous when FromIterator is imported, because Opt doesn't implement that.

---

Niko [said](https://rust-lang.zulipchat.com/#narrow/stream/268952-edition-2021/topic/'Manual.20crater.20run'.20results/near/244167665):

> shouldn't be that hard, we just need a call to `evaluate_predicate` I think

**Comment by m-ou-se** [comment]
(If we don't end up doing this, we should at least change the wording to say `"might become ambiguous"` instead of `"will .."`.)

**Comment by m-ou-se** [comment]
> shouldn't be that hard, we just need a call to evaluate_predicate I think

@nikomatsakis Which exact function on which type do you mean? Do you have some pointers here?

**Comment by jyn514** [comment]
@m-ou-se Maybe Niko meant evaluate-obligation https://doc.rust-lang.org/nightly/nightly-rustc/rustc_trait_selection/traits/query/evaluate_obligation/trait.InferCtxtExt.html#tymethod.evaluate_obligation ?

**Comment by m-ou-se** [comment]
Okay I have an implementation that uh.. compiles. And seems to work, maybe? (:
